trin-core/src/cli.rs

```diff
@@ -180,18 +218,28 @@ impl TrinConfig {
         I: Iterator<Item = T>,
         T: Into<OsString> + Clone,
     {
-        let config = Self::from_iter(args);
+        let config = match Self::from_iter_safe(args) {
```

Comment:
I'm not sure I agree with this approach. If a user provides invalid flags, it seems to me like we should panic, and tell them explicitly what went wrong, rather than silently falling back to using the default flags

Reply:
I agree that this is not the perfect solution. See my previous comment. 
I wasn't sure if we should panic, but it makes sense to panic to ensure that the user's provided config is used.

Reply:
Yup, I agree panic is the way to go here. Seems to me like we can just leave it as `let config = Self::from_iter(args);` which will [panic on failure](https://docs.rs/structopt/latest/structopt/trait.StructOpt.html#method.from_iter).

Reply:
I thought the same first, but it don't really panic, [it prints the error and call `process::exit(1);`](https://docs.rs/clap/2.33.3/clap/struct.App.html#method.get_matches_from).
As we handle other cases with panic, it might be better to panic here as well, so we can use `should_panic` in tests.

Reply:
> Yup, I agree panic is the way to go here. Seems to me like we can just leave it as `let config = Self::from_iter(args);` which will [panic on failure](https://docs.rs/structopt/latest/structopt/trait.StructOpt.html#method.from_iter).

I thought the same first, but it don't really panic, [it prints the error and call process::exit(1);](https://docs.rs/clap/2.33.3/clap/struct.App.html#method.get_matches_from).
As we handle other cases with panic, it might be better to panic here as well, so we can use should_panic in tests.

Reply:
Ahh, gotcha. Yeah, it seems like we panic on those other cases explicitly inside `from_cli()`. But, in this case it makes sense to use structopt's native type casting. Could we update it to something like...
```
// Panic on invalid unwrap of `from_iter_safe()` to force panic for testing purposes.
let config = Self::from_iter_safe(args).unwrap_or_else(|e| panic!("{e}"));
```

Reply:
Thank you for the rust syntax, it's much cleaner!